src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,338 @@
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        public void ApplyStop()
+        {
+            if (IsEnabled()) WriteEvent(2);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to the setup for a BuildManager to receive build requests.
+        /// </summary>
+        public void BuildStart()
+        {
+            if (IsEnabled()) WriteEvent(3);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        public void BuildStop()
+        {
+            if (IsEnabled()) WriteEvent(4);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information of how a project file built.
+        /// </summary>
+        public void BuildProjectStart()
+        {
+            if (IsEnabled()) WriteEvent(5);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        public void BuildProjectStop()
+        {
+            if (IsEnabled()) WriteEvent(6);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to resolving references.
+        /// </summary>
+        public void ComputeClosureStart()
+        {
+            if (IsEnabled()) WriteEvent(7);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        public void ComputeClosureStop()
+        {
+            if (IsEnabled()) WriteEvent(8);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to resolving conditionals.
+        /// </summary>
+        /// <param name="condition">Relevant information about where in the run of the progam it is.</param>
+        public void EvaluateConditionStart(string condition)
+        {
+            if (IsEnabled()) WriteEvent(9, condition);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of the end of the specified event.
+        /// </summary>
+        /// <param name="condition">Relevant information about where in the run of the progam it is.</param>
+        public void EvaluateConditionStop(string condition)
+        {
+            if (IsEnabled()) WriteEvent(10, condition);
+        }
+
+        /// <summary>
+        /// Call this method to notify listeners of how the project data was evaluated.
+        /// </summary>
+        /// <param name="file">Relevant information about where in the run of the progam it is.</param>
+        public void EvaluateStart(string file)
+        {
+            if (IsEnabled())
+            {
+                string projectFile = String.IsNullOrEmpty(file) ? "(null)" : file;
```

Comment:
You should not need this.  You can detect a null or empty argument at analysis time.  NULL values are valid values in traces.  Also, if you want to do processing of arguments it needs to happen outside of `[Event]` methods, either in a `[NonEvent]` method or at the callsite itself.